devito/propagator.py

```diff
@@ -342,10 +342,11 @@ def sympy_to_cgen(self, stencils):
             stencil = self.convert_equality_to_cgen(equality)
             stmts.append(stencil)
 
-        nest_decl = lambda dec, stmt: cgen.Assign(dec.__str__()[:-1], stmt.rvalue)
+        def simplify(dec, stmt):
```

Comment:
Why even define this? The body of this function is now small enough to just insert it into the loop directly.